servo/connectors/kubernetes.py

```diff
@@ -2539,9 +3124,14 @@ class CanaryOptimization(BaseOptimization):
 
     @classmethod
     async def create(
-        cls, deployment_config: "DeploymentConfiguration", **kwargs
+        cls, deployment_config: Union["DeploymentConfiguration", "RolloutConfiguration"], **kwargs
```

Comment:
Rename away from `deployment_config` since it can now be something other than a Deployment.